### PR TITLE
Add try/catch block

### LINKED ISSUE #344: unhandled promise rejection
When I try to run some commands with a wrong user, the `clasp` shows error messages caused by problems with the permissions but remains frozen.

## Expected Behavior

Trying to run a command, `clasp` must show an error message and end.

```
$ clasp logs --json
Error: User has not been authorized
(end)
```

*NOTE:* This happens also with other commands such as `clasp deploy`.

## Actual Behavior

The `clasp` application shows an error...

```
 clasp logs --json
- Using local credentials: ./clasprc.json 🔐 Grabbing logs...(node:16843) UnhandledPromiseRejectionWarning: Error: The caller does not have permission
    at createError (/usr/lib/node_modules/@google/clasp/node_modules/axios/lib/core/createError.js:16:15)
    at settle (/usr/lib/node_modules/@google/clasp/node_modules/axios/lib/core/settle.js:18:12)
    at Unzip.handleStreamEnd (/usr/lib/node_modules/@google/clasp/node_modules/axios/lib/adapters/http.js:201:11)
    at emitNone (events.js:111:20)
    at Unzip.emit (events.js:208:7)
    at endReadableNT (_stream_readable.js:1064:12)
    at _combinedTickCallback (internal/process/next_tick.js:139:11)
    at process._tickCallback (internal/process/next_tick.js:181:9)
(node:16843) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:16843) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
/ Using local credentials: ./clasprc.json 
🔐 Grabbing logs...
```

but remains frozen. I must stop the `clasp` using `[Ctrl]+[C]`

## Steps to Reproduce the Problem

  1. Create an App Script project where users can see the project if they have the link
  1. Login using `clasp`:  `clasp login`
  1. Clone/Pull the content of the above mentioned project:  `clasp clone ....`
  1. Run commands such as `clasp deploy` or `clasp login --json`

## Specifications

  - Node version (`node -v`): v10.11.0  
  - Version (`clasp -v`): 1.6.3
  - OS (Mac/Linux/Windows): Linux -- Ubuntu 18.04.1 LTS

### PR DESCRIPTION
**Pull request by takanakahiko**
Fixes #344 

Wrap process accessing API by `try/catch` block.

- [x] `npm run test` succeeds.
- [x] `npm run lint` succeeds.
- [x] Appropriate changes to README are included in PR.

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -386,4 +386,12 @@ export async function validateManifest(): Promise<boolean> {
     return false;
   }
   return true;
-}
\ No newline at end of file
+}
+
+/**
+ * Error handling when accessing script api.
```
Comment (grant) [inline src/utils.ts]: Handles error when making an API request.

**Comment by takanakahiko** [comment]
fixed

**Comment by grant** [comment]
Actually, a better design would be something like this:

`utils.ts`
```ts
/**
 * Handles error when making an API request.
 */
export function scriptApiErrorHandler(method:Function): void {
  try {
    method();
  } catch (e) {
    spinner.stop(true);
    return logError(null, e.message);
  }
}
```

Then wrap each method with a error handler function:

`commands.ts`
```js
commander
  .command('clone [scriptId] [versionNumber]')
  .description('Clone a project')
  .action(clone);

// ->

commander
  .command('clone [scriptId] [versionNumber]')
  .description('Clone a project')
  .action(handleAPIError(clone));
```

Then we don't need any `try/catch` blocks in our `commands.ts`.
What do you think?

**Comment by grant** [comment]
I think you may be right:

```ts
 .action((...args) => handleAPIError(clone, args));
```

**Comment by takanakahiko** [comment]
I want to implement it according to this comment .

https://github.com/tj/commander.js/issues/505#issuecomment-244988124


```
/**
 * Handles error when making an API request.
 */
export function scriptApiErrorHandler(command: (...args) => Promise<void>) {
  return async (...args: any[])=> {
    try {
      await command(...args);
    } catch (e) {
      spinner.stop(true);
      logError(null, e.message);
    }
  }
}
```

```
commander
  .command('clone [scriptId] [versionNumber]')
  .description('Clone a project')
  .action(handleAPIError(clone));
```

What do you think?

**Comment by grant** [comment]
That plan looks great.

**Comment by takanakahiko** [comment]
Thanks! 
I will close this and make a another PR.

**Pull request closed without merging**